### DISCUSSION TITLE
Tiny VGG performance on Pizza_Steak_Sushi Dataset

### DISCUSSION THREAD

**Discussion by Jahanpanah1994MD** [mrdbourke/pytorch-deep-learning#327] [Q&A]
Hi
 I just finished the PyTorch video on Youtube, but I can't still figure out how we can increase accuracy to an acceptable level. I tried different methods that Daniel mentioned in the video, but none of them worked properly. My model is still as good as guessing. So, how we should do it?? Is there another video that explains it?

**Comment by mrdbourke** [comment]
Hi @Jahanpanah1994MD,

Which notebook are you working through? 

What have you tried so far? 

Have you been through the steps here: https://www.learnpytorch.io/02_pytorch_classification/#5-improving-a-model-from-a-model-perspective 

One of the most important things when training neural networks is to experiment, experiment, experiment.

Not everything will always work.

For one way to really improve your model's performance, check out transfer learning: https://www.learnpytorch.io/06_pytorch_transfer_learning/

**Reply by Jahanpanah1994MD** [reply]
Hi Daniel,
Thank you for responding,
Yes I should start the transfer learning section.  
I tried changing the number of hidden units,layers and learning rate, I also trained the model for longer, however they didn't help much.
Thx again.

**Comment by kmoruss** [comment]
I'm in the same boat...will look at transfer learning. I'm about to start training on my own images and want to make sure I don't run into the same roadblocks of not being able to improve it.

**Comment by kmoruss** [comment]
Additionally to changing hidden units, learning rate, epochs, I also played with batch size and image resize size

**Reply by shisirkha** [reply]
Use more unique layers like dropout and batchnorm2d layers

**Comment by kmoruss** [comment]
![Section_04_results](https://github.com/mrdbourke/pytorch-deep-learning/assets/67767972/71904923-4a27-429d-980b-74f571ca931e)
I reduced learning rate, increased epochs to 30 and slowly Tiny VGG also became little better, then I added new model_2 based on AlexNet architecture and with very small learning rate and 30 epochs it was able to achieve 70% accuracy.

I also downloaded full food101 dataset for pizza, steak and sushi where I got to use 800 images for training and 200 for testing.

What is odd to me is that model_0 is able to achieve 100% training accuracy but when it comes to actually predicting something i.e. Testing accuracy is still low.

**Reply by shisirkha** [reply]
HOW YOU GOT 100% accuracy

**Comment by uros1610** [comment]
I've tried a lot of things, from messing with the hyperparameters, to expanding my dataset 5-6 times with randomly rotated, blurred, and cropped images, none of which got me test accuracy of over 60%, but the train accuracy is always around 90-95%. I tried adding Dropout layers with probability = 0.5, which helped a bit but still the model overfits the training data A LOT.
